src/ynEIGEN/AssetRegistry.sol

```diff
@@ -0,0 +1,342 @@
+     */
+    function disableAsset(IERC20 asset)
+    public
+    onlyRole(ASSET_MANAGER_ROLE)
+    notZeroAddress(address(asset))
+    whenNotPaused {
+        if (!_assetData[asset].active) {
+            revert AssetAlreadyInactive(address(asset));
+        }
+
+        _assetData[asset].active = false;
+
+        emit AssetDeactivated(address(asset));
+    }
+
+    /**
+     * @notice Deletes an asset from the system entirely.
+     * @dev Removes an asset from the _assetData mapping and the assets array. This function can only be called by the strategy manager.
+     * @param asset The address of the ERC20 token to be deleted.
+     */
+    function deleteAsset(IERC20 asset)
+    public
+    onlyRole(ASSET_MANAGER_ROLE)
+    notZeroAddress(address(asset))
+    whenNotPaused {
+        if (!_assetData[asset].active) {
+            revert AssetNotActiveOrNonexistent(address(asset));
+        }
+
+        uint256 balanceInPool = asset.balanceOf(address(this));
+        if (balanceInPool != 0) {
+            revert AssetBalanceNonZeroInPool(balanceInPool);
+        }
+
+        uint256 strategyBalance = eigenStrategyManager.getStakedAssetBalance(asset);
+        if (strategyBalance != 0) {
+            revert AssetBalanceNonZeroInStrategyManager(strategyBalance);
+        }
+
+        // Remove asset from the assets array
+        uint256 assetIndex = findAssetIndex(asset);
+
+        // Move the last element into the place to delete
+        assets[assetIndex] = assets[assets.length - 1];
+        assets.pop();
+
+        // Remove asset from the mapping
+        delete _assetData[asset];
+
+        emit AssetDeleted(address(asset), assetIndex);
+    }
+
+    /**
+     * @notice Finds the index of an asset in the assets array.
+     * @param asset The asset to find.
+     * @return uint256 The index of the asset.
+     */
+    function findAssetIndex(IERC20 asset) internal view returns (uint256) {
+        for (uint256 i = 0; i < assets.length; i++) {
+            if (assets[i] == asset) {
+                return i;
+            }
+        }
+        revert AssetNotFound(address(asset));
+    }
+
+    //--------------------------------------------------------------------------------------
+    //----------------------------------  ASSET VALUE  -------------------------------------
+    //--------------------------------------------------------------------------------------
+
+    /**
+     * @notice This function calculates the total assets of the contract
+     * @dev It iterates over all the assets in the contract, gets the latest price for each asset from the oracle, 
+     * multiplies it with the balance of the asset and adds it to the total
+     * @return total The total assets of the contract in the form of uint
+     */
+    function totalAssets() public view returns (uint256) {
+        uint256 total = 0;
+
+        uint256[] memory depositedBalances = getAllAssetBalances();
+
+        for (uint256 i = 0; i < assets.length; i++) {
```

Comment:
cache the length before the loop to save gas: `uint256 assetsLength = assets.length`